governance/program/src/state.rs

```diff
@@ -0,0 +1,247 @@
+//! Program accounts
+
+use solana_program::pubkey::Pubkey;
+
+/// Max length of a governance name
+pub const MAX_GOVERNANCE_NAME_LENGTH: usize = 32;
+
+/// Max length of a proposal description link
+pub const MAX_PROPOSAL_DESCRIPTION_LINK_LENGTH: usize = 200;
+
+/// Max length of a proposal name
+pub const MAX_PROPOSAL_NAME_LENGTH: usize = 32;
+
+/// Max length of a proposal's instruction data
+pub const MAX_PROPOSAL_INSTRUCTION_DATA_LENGTH: usize = 450;
+
+/// Max number of transactions allowed for a proposal
+pub const MAX_TRANSACTIONS: usize = 5;
+
+/// Defines all Governance accounts types
+#[derive(Clone, Debug, PartialEq)]
+pub enum GovernanceAccountType {
+    /// 0 - Default uninitialized account state
+    Uninitialized,
+
+    /// 1 - Governance account
+    Governance,
+
+    /// 2 - Proposal account for Governance account. A single Governance account can have multiple Proposal accounts
+    Proposal,
+
+    /// 3 - Proposal voting state account. Every Proposal account has exactly one ProposalState account
+    ProposalState,
```

Comment:
Let's hope that by removing all of those extra mints, you can actually fit the whole proposal in one struct, rather than splitting it up

Reply:
Ok, I kept it for now but I'll get back to it so see if we can merge the accounts.